base/collection_n1ql_common.go

```diff
@@ -128,7 +132,32 @@ func (im *indexManager) GetAllIndexes() ([]gocb.QueryIndex, error) {
 //	  CreateIndex("myIndex", "field1, field2, nested.field", "field1 > 0", N1qlIndexOptions{numReplica:1})
 //	CREATE INDEX myIndex on myBucket(field1, field2, nested.field) WHERE field1 > 0 WITH {"numReplica":1}
 func CreateIndex(ctx context.Context, store N1QLStore, indexName string, expression string, filterExpression string, options *N1qlIndexOptions) error {
-	createStatement := fmt.Sprintf("CREATE INDEX `%s` ON %s(%s)", indexName, store.EscapedKeyspace(), expression)
+	return createIndex(ctx, store, indexName, expression, filterExpression, false, options)
+}
+
+// CreateIndexIfNotExists issues a CREATE INDEX query in the current bucket, using the form:
```

Comment:
This is more accurately create an index in the current collection, I think?

Reply:
Yep, old comment stolen from `CreateIndex`. Have removed all bucket references.